tests/sgmcmc/test_sgnht.py

```diff
@@ -0,0 +1,85 @@
+from functools import partial
+import torch
+from optree import tree_map
+from optree.integration.torch import tree_ravel
+
+from posteriors.sgmcmc import sgnht
+
+from tests.scenarios import batch_normal_log_prob
+
+
+def test_sgnht():
+    torch.manual_seed(42)
+    target_mean = {"a": torch.randn(2, 1) + 10, "b": torch.randn(1, 1) + 10}
+    target_sds = tree_map(lambda x: torch.randn_like(x).abs(), target_mean)
+
+    target_mean_flat = tree_ravel(target_mean)[0]
+    target_cov = torch.diag(tree_ravel(target_sds)[0] ** 2)
+
+    batch = torch.arange(10).reshape(-1, 1)
+
+    batch_normal_log_prob_spec = partial(
+        batch_normal_log_prob, mean=target_mean, sd_diag=target_sds
+    )
+
+    n_steps = 10000
+    lr = 1e-2
+    alpha = 1.0
+    beta = 0.0
+
+    params = tree_map(lambda x: torch.zeros_like(x), target_mean)
+    init_params_copy = tree_map(lambda x: x.clone(), params)
+
+    sampler = sgnht.build(batch_normal_log_prob_spec, lr=lr, alpha=alpha, beta=beta)
+
+    # Test inplace = False
+    sgnht_state = sampler.init(params)
+    log_posts = []
+    all_params = tree_map(lambda x: x.unsqueeze(0), params)
+
+    for _ in range(n_steps):
+        sgnht_state = sampler.update(sgnht_state, batch, inplace=False)
+
+        all_params = tree_map(
+            lambda x, y: torch.cat((x, y.unsqueeze(0))), all_params, sgnht_state.params
+        )
+
+        log_posts.append(sgnht_state.log_posterior.item())
+
+    burnin = 5000
```

Comment:
is there an explanation for why burnin has to be larger than in the other tests?

Reply:
Not really haha, tbh I think it should be larger for the other tests too